aten/src/ATen/native/quantized/cpu/qnnpack_utils.h

```diff
@@ -84,3 +89,58 @@ inline std::pair<uint8_t, uint8_t> activationLimits(
   }
 }
 #endif
+
+namespace {
+std::vector<float> generate_requantization_scales(
+    const at::Tensor& weight_scales,
+    const float input_scale,
+    const float output_scale) {
+  auto num_output_channels = weight_scales.numel();
+  float* weight_scales_data = weight_scales.data_ptr<float>();
+  std::vector<float> requant_scales(num_output_channels, 1.f);
+  for (int i = 0; i < num_output_channels; ++i) {
+    requant_scales[i] = weight_scales_data[i] * input_scale / output_scale;
+  }
+  return requant_scales;
+}
+
+std::pair<at::Tensor, at::Tensor> make_zero_points_and_scales_tensor(
+    const at::Tensor& weight_contig
+    ) {
+  auto num_output_channels = weight_contig.size(0);
+  const auto qtype = weight_contig.qscheme();
+  at::Tensor weight_zp =
+    at::native::empty_cpu({1}, at::device(at::kCPU).dtype(at::kQUInt8));
+  // Adjust weight zero point, similar to weight data.
+  uint8_t* weight_zp_data;
+  if (qtype == at::kPerTensorAffine) {
+    weight_zp_data = (uint8_t*)weight_zp.data_ptr<c10::quint8>();
```

Comment:
Prefer reinterpret_cast over c style cast

Reply:
Sure, but any particular reason or stylistic? I can make the change.

Reply:
I think it is better to be explicit in the type of cast we are using. Also general convention in these files.

Reply:
That sounds good.

Reply:
I think it would be good to address the c style cast used everywhere in this PR.

Reply:
Definitely. Will do that.